src/zmalloc.c

```diff
@@ -415,7 +457,13 @@ char *zstrdup(const char *s) {
 }
 
 size_t zmalloc_used_memory(void) {
-    size_t um = atomic_load_explicit(&used_memory, memory_order_relaxed);
+    size_t um = 0;
+    for (int i = 0; i < total_active_threads; i++) {
+        um += used_memory_thread[i];
```

Comment:
Now total_active_threads can be greater than MAX_THREADS_NUM so...

```suggestion
    for (int i = 0; i < min(total_active_threads, MAX_THREADS_NUM); i++) {
        um += used_memory_thread[i];
```